### PR TITLE
refactor(nextjs-template): general comments small changes

### PR DESCRIPTION
**Pull request by hkievet**
# Explanation

I had a hard time groking the build process for the nextjs application.  So I made some changes that should make things easier to grok in the future.

Notably:
* lots of comments
* lots of typings
* removing unused dependencies

Perhaps easiest to review going commit by commit?

Making this PR before the changeset gets any larger.

# Dendron Extended PR Checklist

- NOTE: the links don't work. you'll need to go into the wiki and use lookup to find the note until we fix some issues in the markdown export

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [ ] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
  - [ ] check whether code be simplified
  - [ ] check if similar function already exist in the codebase. if so, can it be re-used?
  - [ ] check if this change adversely impact performance
- Operations
  - [ ] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [ ] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended
- [ ] can we track the performance of this change to know if it is _successful_?
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

### Basics

- [ ] [Write Tests](dev.process.qa.test) 
- [ ] [Confirm existing tests pass](dev.process.qa.test)
- [ ] [Confirm manual testing](dev.process.qa.test) 
- [ ] Common cases tested
- [ ] 1-2 Edge cases tested
- [ ] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

### Extended
- [ ] If you are adding a new language feature (graphically visible in VS Code/preview/publishing), an example is included in the [test workspace](dev.ref.test-workspace)
- CSS
  - [ ] display is correct for following dimensions
    - [ ] sm: screen ≥ 576px, eg. iphonex, (375x812)
    - [ ] lg: screen ≥ 992px
    - [ ] xxl: screen ≥ 1600px eg. mac (1600x900)
  - [ ] display is correct for following browsers (across the various dimensions)
    - [ ] safari
    - [ ] firefox
    - [ ] chrome

## Docs
- [ ] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
- [ ] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

### Extended
- [ ]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)
  - eg: [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

File: packages/nextjs-template/components/DendronCollection.tsx [hunk 1]
```diff
@@ -57,17 +55,18 @@ export function prepChildrenForCollection(
...
-function ISO2FormattedDate(time: string, format: Intl.DateTimeFormatOptions) {
+function ISO2FormattedDate(
+  time: string,
+  format: Intl.DateTimeFormatOptions,
+): string {
   const dt = DateTime.fromISO(time);
   return dt.toLocaleString(format);
 }
 
-function millisToJSDate(ts: number) {
```
Comment (hkievet) [inline packages/nextjs-template/components/DendronCollection.tsx:65]: Unused

File: packages/nextjs-template/pages/notes/[id].tsx [hunk 2]
```diff
@@ -16,16 +17,21 @@ import {
 
 export default DendronNotePage;
 
-export const getStaticPaths: GetStaticPaths = getNotePaths;
+export const getStaticPaths: GetStaticPaths<DendronNotePageParams> =
+  getNotePaths;
 
-export const getStaticProps: GetStaticProps = async (
-  context: GetStaticPropsContext
+export const getStaticProps: GetStaticProps<
```
Comment (hkievet) [inline packages/nextjs-template/pages/notes/[id].tsx:23]: See the following URL for a typing explanation.
https://www.vitamindev.com/next-js/getstaticprops-getstaticpaths-typescript/

File: packages/nextjs-template/tsconfig.build.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
     "esModuleInterop": true,
     "strict": true,
     "declaration": true,
-    "types": ["node", "jest"],
+    "types": ["node"],
```
Comment (hkievet) [inline packages/nextjs-template/tsconfig.build.json:13]: Unused types, we don't use jest

File: packages/nextjs-template/utils/build.ts [hunk 4]
```diff
@@ -32,48 +36,65 @@ export function getNoteBody(id: string) {
...
+export function getNoteMeta(id: string): Promise<NoteProps> {
   const dataDir = getDataDir();
   return fs.readJSON(
-    path.join(dataDir, NOTE_META_DIR, `${id}.json`)
-  ) as Promise<NoteProps>;
+    path.join(dataDir, NOTE_META_DIR, `${id}.json`),
+  );
 }
 
+let _CONFIG_CACHE: IntermediateDendronConfig | undefined;
```
Comment (hkievet) [inline packages/nextjs-template/utils/build.ts:84]: getConfig() is called a lot during the build process... multiple times for each note

**Review by kevinslin — APPROVED** [review]

**Pull request merged**